_posts/2022-02-04-solidity-developer-survey-2021-results.md

```diff
@@ -0,0 +1,322 @@
+---
+layout: post
+published: true
+title: 'Solidity Developer Survey 2021 Results'
+date: '2022-02-03'
+author: Franziska Heintel
+category: Announcements
+---
+
+In this post, we will be summarizing and analyzing the results of the [2021 Solidity Developer Survey](https://blog.soliditylang.org/2021/11/18/solidity-developer-survey-2021/).
+
+A big thank you goes out to everybody who took the time and participated!
+
+Your input is invaluable to us and plays a crucial role in helping to continuously improve the Solidity developer experience as a whole.
+
+## Summary & Notable Insights
+
+- **Survey Audience**: In total, 435 developers from 73 different countries participated in the 2021 survey. Compared to 2020, that is more than a 100% increase in responses. The coverage of different geographies also drastically increased from 48 countries in 2020 to 73 countries in 2021. More than 20% stated to be residing in the US, followed by India (9%) and Germany (4%). Roughly 9% preferred to not share details on their location.
+- **Developer Profiles**: The level of coding experience is medium to high with the majority of respondents having coded professionally 3 or more years, 36.6% even more than 6 years.
+- **Solidity Experience**: More than half of all respondents have been using Solidity less than a year, while 15.5% have been using it more than 3 years. Almost 80% use Solidity daily or weekly.
+- **Solidity Expertise**: Most respondents deem themselves Solidity experts, with a self rating in expertise of 7 or higher (scale of 10). 4.2% rate their expertise as a 10/10. 80% of respondents use Solidity for their personal projects, roughly 60% also use it at work. More than half of all respondents have been using Solidity for less than a year.
+- **Developer Experience**: The majority (+70%) believes that the Solidity developer experience improved in the last year. Only 1.6% are of the opinion it got worse.
+- **Language Restrictiveness**: More than 60% of respondents wish that Solidity becomes more restrictive/explicit (having more checks). 26% would prefer it to remain as is.
+- **Future Features**: A more efficient optimizer and the ability to catch custom errors are ranked as most important future features under discussion. Moreover, floating point support, better array management and fixing stack too deep are among the most mentioned anticipated features.
+- **Liked & Dreaded Features**: Respondents most like Solidity's simplicity, the "easy to learn" aspect of it, SafeMath by default and modifiers. Dreaded topics are debugging, the stack too deep error and missing support for floating points.
```

Comment:
```suggestion
- **Liked & Dreaded Features**: Respondents most like Solidity's simplicity, the "easy to learn" aspect of it, SafeMath by default and modifiers. Dreaded topics are debugging, the stack too deep error and missing support for floating point numbers.
```
maybe

Reply:
or types...

Reply:
Also generally not sure if we want to stick to "floating point", since that's what people literally put into the answers, or change it to "fixed points", since that's what we suppose they actually mean :-). Maybe better to stick with the actual data, but maybe also worth a note somewhere...

Reply:
Maybe in ``[3] Features`` a note like that we suspect "floating point" to really mean "fixed point", but that we'd welcome clarification on that or something....